### PR TITLE
Use jstree

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR modifies the tree HTML representation to use jstree. Also adds a "level" parameter to tree() to allow limiting depth to which tree is printed.

### CODE REVIEW

**Comment by alimanfoo** [comment]
[example notebook](http://nbviewer.jupyter.org/github/alimanfoo/zarr/blob/5cc474a3ff21afc7b0173aacb13aa7aae1b4c6c8/notebooks/repr_tree.ipynb) is working in nbviewer

**Comment by alimanfoo** [comment]
cc @jakirkham

File: notebooks/repr_tree.ipynb [hunk 1]
```diff
@@ -6,13 +6,9 @@
...
    "source": [
-    "import zarr\n",
-    "\n",
-    "# Python 2/3 trick\n",
-    "try:\n",
-    "    unicode\n",
-    "except NameError:\n",
-    "    unicode = str"
+    "import sys\n",
+    "sys.path.insert(0, '..')\n",
```
Comment (jakirkham) [inline notebooks/repr_tree.ipynb]: Why is this needed?

File: zarr/util.py [hunk 2]
```diff
@@ -312,117 +313,126 @@ def _repr_html_(self):
...
-    """)
-    result += "</style>\n\n"
-
-    # Insert the HTML list
-    result += """<div class="zarr-tree">\n"""
-    result += "<ul>\n"
-    result += custom_html_sublist(group, indent=indent)
-    result += "</ul>\n"
-    result += "</div>\n"
+def tree_html(group, expand, level):
```
Comment (jakirkham) [inline zarr/util.py]: This is probably my error in the first place, but should we just make this a method of `TreeViewer`?

File: zarr/util.py [hunk 3]
```diff
@@ -312,117 +313,126 @@ def _repr_html_(self):
...
+    """ % (node_id, tree_group_icon, tree_array_icon))
 
     return result
 
 
+tree_group_icon = 'fa fa-folder'
+tree_array_icon = 'fa fa-table'
+# alternatives...
+# tree_group_icon: 'jstree-folder'
+# tree_array_icon: 'jstree-file'
```
Comment (jakirkham) [inline zarr/util.py]: Maybe these should be variables in `TreeViewer`.

File: notebooks/repr_tree.ipynb (hunk 1 above)
Comment (alimanfoo) [reply notebooks/repr_tree.ipynb]: So zarr is on the path. How did you do that, did you set PYTHONPATH?

File: zarr/util.py [hunk 4]
```diff
@@ -312,117 +313,126 @@ def _repr_html_(self):
...
-    result += custom_html_sublist(group, indent=indent)
-    result += "</ul>\n"
-    result += "</div>\n"
+def tree_html(group, expand, level):
+
+    result = ''
+
+    # include CSS for jstree default theme
+    css_url = '//cdnjs.cloudflare.com/ajax/libs/jstree/3.3.3/themes/default/style.min.css'
+    result += '<link rel="stylesheet" href="{}"/>'.format(css_url)
```
[the remark this answers is not in the document]
Comment (alimanfoo) [reply zarr/util.py]: I think that just gets handled automatically. I've just been copying examples there.

**Comment by jakirkham** [comment]
Looks pretty good to me. Made a few comments above, but nothing essential. Would be nice to get rid of the `sys.path` hack if possible, but not crucial if there is some issue requiring it as a workaround.

File: zarr/util.py (hunk 2 above)
Comment (alimanfoo) [reply zarr/util.py]: Yes, could do.

File: zarr/util.py (hunk 3 above)
Comment (alimanfoo) [reply zarr/util.py]: Yes could be. I was trying to think of best way to make it so a really enthusiastic user could change the default icons if they want to use their favourite emojis instead :-)

**Comment by alimanfoo** [comment]
Thanks for review, would be good to know what you do instead of sys.path hack.

File: zarr/util.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jakirkham) [reply zarr/util.py]: Guess so. Interesting. Also this [SO question]( https://stackoverflow.com/q/9374203 ) seems relevant.

File: zarr/util.py (hunk 3 above)
Comment (jakirkham) [reply zarr/util.py]: Was thinking about having static variables of the class as an option as well. Maybe that would work better?

**Comment by jakirkham** [comment]
Normally I've been doing `pip install -e .` when working locally, which ensures the development version is on the path. If that's not working, would be curious to know more details about the issue that is causing the need for the hack.

**Comment by alimanfoo** [comment]
Thanks for the tip re pip install -e, I've updated the demo notebook. Also added support for ``expand`` being an int, so can control to what depth tree is expanded. I think I'll leave the rest as-is, having some of these methods broken out as functions is convenient, especially where long lines are needed for embedded html/javascript.

**Comment by jakirkham** [comment]
Thanks @alimanfoo. Sounds reasonable. We can always go back and tweak later if needed.
